cpp/cmake_modules/FindClangTools.cmake

```diff
@@ -132,15 +130,14 @@ else()
       clang-format-3.7
       clang-format-3.6
       clang-format
```

Comment:
Aside: we probably don't want to accept a clang-format version other than the pinned version, so these other versions can be removed

Reply:
Those are in the else block of if (clang-format version provided), so I guess we should error out if clang-format's version isn't provided

Reply:
no failing to find clang-format would be better I suppose